flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultPreferredLocationsRetriever.java

```diff
@@ -35,7 +35,7 @@
 
 /**
  * Default implementation of {@link PreferredLocationsRetriever}. Locations based on state will be
- * returned if exist. Otherwise locations based on inputs will be returned.
+ * returned if existed. Otherwise, locations based on inputs will be returned.
```

Comment:
should be "exists"

Reply:
The usage of `exist` is correct regarding `locations`.

And I would suggest to not make this kind of changes, if the statement was not misleading or incorrect, unless you need to update the statement because it is outdated.